src/stactools/sentinel2/product_metadata.py

```diff
@@ -16,6 +16,22 @@ class ProductMetadataError(Exception):
     pass
 
 
+def fix_z_values(coord_values: List[str]) -> List[float]:
+    """Some geometries have a '0' value in the z position
+    of the coordinates. This method detects and removes z
+    postion coordinates. This assumes that in cases where
+    the z value is included, it is included for all coordinates.
+    """
+    if len(coord_values) % 3 == 0:
+        # Check if all of the 3rd position values are '0'
+        if not any(
+            [x
+             for i, x in enumerate(coord_values) if i % 3 == 2 and x != '0']):
+            return [float(c) for c in coord_values if c and c != '0']
```

Comment:
In the case where there's x or y `'0'` values, won't this corrupt the coordinate array?

Reply:
Yes, if there are both all 0 z values and a 0 in x or y, it will mess things up; this was assuming those cases wouldn't intersect but obviously my assumptions about this aren't tending towards correct :-D I'll change to only remove 3rd position 0's.